blueocean-core-js/src/js/capability/CapabilityAugmenter.js

```diff
@@ -0,0 +1,99 @@
+/**
+ * Created by cmeyers on 9/8/16.
+ */
+
+const addClass = (clazz, classMap) => {
+    const className = clazz._class;
+
+    if (!classMap[className]) {
+        // eslint-disable-next-line no-param-reassign
+        classMap[className] = [];
+    }
+
+    classMap[className].push(clazz);
+};
+
+const canWalk = (item) => item && (typeof item === 'object' || Array.isArray(item));
+
+// TODO: can we really filter out 'actions' in all cases? should it use a leading underscore?
+const IGNORED_PROPS = ['_links', 'actions'];
```

Comment:
https://github.com/jenkinsci/blueocean-plugin/blob/master/blueocean-dashboard/src/main/js/util/logDisplayHelper.js#L35

```
        // FIXME: TS I need to talk to cliffMeyers how we can refactor the following code to use capabilities
        // the problem I see ATM is that we would need to ask the c-API every time for each action, whether this
        // action has the capability for logging
        const hasLogs = item.actions ? item.actions
                .filter(action => action._class === 'org.jenkinsci.plugins.workflow.support.actions.LogActionImpl').length > 0
            : false;
```

If we filter this action we lose knowledge as the above, or?